src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
>listing all exceptions explicitly is unmaintainable

Fair, will remove it then.

>@throws Exception in the DecoderInterface does not bring anything useful IMO.

I beg to differ, if you're typehinting against interface and no `@throws` tag is given in one of its method, it's only fair to not expect an exception throw by it (unless you really messed something up). Why `DecoderInterface` shouldn't benefit from the `@throws` tag as any other class? It's not less valuable or critical than another interface. Also not having it make it only confusing if you're implementing your own decoder: I am allowed to throw an exception? Must it be a specific one? Not adding this `@throws` condemn you to check how other decorders or the `Serializer` works to find an answer.